build/e2e_test.go

```diff
@@ -600,3 +603,233 @@ func TestEmptyContent(t *testing.T) {
+
+		fooAtMainAndRelease = zoekt.Document{Name: "foo.go", Branches: []string{"main", "release"}, Content: []byte("common foo-main-and-release")}
+
+		barAtMain   = zoekt.Document{Name: "bar.go", Branches: []string{"main"}, Content: []byte("common bar-main")}
+		barAtMainV2 = zoekt.Document{Name: "bar.go", Branches: []string{"main"}, Content: []byte("common bar-main-v2")}
+
+		bazAtMain = zoekt.Document{Name: "baz.go", Branches: []string{"main"}, Content: []byte("common baz-main")}
+	)
+
+	for _, test := range []struct {
+		name  string
+		steps []step
+	}{
+		{
+			name: "tombstone older documents",
+			steps: []step{
+				{
+					name:              "setup",
+					documents:         []zoekt.Document{barAtMain, fooAtMain},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMain, fooAtMain},
+				},
+				{
+					name:      "add new version of foo, tombstone older ones",
+					documents: []zoekt.Document{fooAtMainV2},
+					optFn: func(t *testing.T, o *Options) {
+						o.IsDelta = true
+						o.ChangedOrRemovedFiles = []string{"foo.go"}
+					},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMain, fooAtMainV2},
+				},
+				{
+					name:      "add new version of bar, tombstone older ones",
+					documents: []zoekt.Document{barAtMainV2},
+					optFn: func(t *testing.T, o *Options) {
+						o.IsDelta = true
+						o.ChangedOrRemovedFiles = []string{"bar.go"}
+					},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMainV2, fooAtMainV2},
+				},
+			},
+		},
+		{
+			name: "tombstone older documents even if the latest shard has no documents",
+			steps: []step{
+				{
+					name:              "setup",
+					documents:         []zoekt.Document{barAtMain, fooAtMain},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMain, fooAtMain},
+				},
+				{
+					// a build with no documents could represent a deletion
+					name:      "tombstone older documents",
+					documents: nil,
+					optFn: func(t *testing.T, o *Options) {
+						o.IsDelta = true
+						o.ChangedOrRemovedFiles = []string{"foo.go"}
+					},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMain},
+				},
+			},
+		},
+		{
+			name: "tombstones affect document across branches",
+			steps: []step{
+				{
+					name:              "setup",
+					documents:         []zoekt.Document{barAtMain, fooAtMainAndRelease},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMain, fooAtMainAndRelease},
+				},
+				{
+
+					name:      "tombstone foo",
+					documents: nil,
+					optFn: func(t *testing.T, o *Options) {
+						o.IsDelta = true
+						o.ChangedOrRemovedFiles = []string{"foo.go"}
+					},
+					query:             "common",
+					expectedDocuments: []zoekt.Document{barAtMain},
+				},
+			},
+		},
+		{
+			name: "tombstones are cumulative",
```

Comment:
Is there a difference between this test at the first one? The only difference I see is that this test doesn't add new documents. Is that relevant to test that tombstones are cumulative?